### PR TITLE
Replace old "country" and "race" name in the code with the generic "faction"

### LINKED ISSUE #7552: Move away from mod-specific engine code (part 86): Countries and Races
As I've mentioned a couple of times before, the `Country` and `Race` are really out-of-place in the engine. They might have been OK once (years ago), but not any more.
You have `Country` and `Race` in the code and in YAML, but the players see factions, countries, alliances of countries, Great Houses, etc.
Currently we have:
 - Tiberian Dawn: GDI vs Nod (government vs terrorists, or faction vs faction)
 - Red Alert: Russia vs USA (country vs country), more popular as USSR vs The Allied Nations (faction vs faction)
 - Dune 2000: Atreides vs Harkonnen vs Ordos (houses) vs Corrino/Emperror (house/faction/government) vs Fremen vs Smugglers vs Mercenaries (factions)

We'll be having also:
 - Tiberian Sun: GDI vs Nod (government vs terrorists, or faction vs faction) (include the Scrin to get a `species`)
 - Firestorm: GDI vs Nod (government vs terrorists, or faction vs faction)
 - Red Alert 2: Russia vs USA (country vs country)
 - Yuri's revenge: Russia vs USA vs Yuri (country vs country vs faction)

Hopefully one day:
 - Generals: USA vs China vs GLA (country vs country vs terrorist faction)
 - Zero Hour: USA vs China vs GLA vs Gen1 vs ... (country vs country vs terrorist faction vs rogue faction 1 vs rogue faction 2 vs ...)

Unofficial mods may include:
 - Civilizations
 - Empires
 - Kingdoms
 - Tribes
 - Races
 - Species

The engine code should be generic enough for all official mods (and as much as possible for any unofficial).
So I am proposing (again) to switch any mentions of `Country`, `Race`, `House`, `Species`, etc. in the code with the generic enough `Faction`.

My earlier statement can be found [here](https://github.com/OpenRA/OpenRA/pull/7399#issuecomment-74166608).

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8964265-move-away-from-mod-specific-engine-code-part-86-countries-and-races?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
I finally got around to doing this.
I replaced all occurrences of "country", "countries", "race" and "races" with "faction" in file names, type names, code and UI-related YAML.
Still need to add **a ton** of YAML upgrade code to upgrade the actor rules.

I will probably squash the commits once everything is done, because they are _a lot_.

Closes  #7552.

### CODE REVIEW

File: OpenRA.Game/Map/PlayerReference.cs [hunk 1]
```diff
@@ -25,9 +25,9 @@ public class PlayerReference
 		public bool AllowBots = true;
 		public bool Required = false;
 
-		public bool LockRace = false;
-		public string Race;
-		public string RaceFlagName;
+		public bool LockFaction = false;
```
Comment (pchote) [inline OpenRA.Game/Map/PlayerReference.cs]: These will need upgrade rules.

**Comment by penev92** [comment]
Depends on #7875 to be able to upgrade maps properly.

**Comment by penev92** [comment]
It's been so long that I don't even remember what the state of this is, but now that #7875 is merged I can finally upgrade ALL! THE! MAPS! (oh, joy!).
And I believe I need a couple more upgrade rules after that.

I will probably be splitting some more parts of this into their own PRs to make reviewers' lifes a bit easier, but those would probably be the missing parts.

**Comment by penev92** [comment]
I am closing this so it doesn't clutter the PR queue.
It will come back, hopefully soon, but I will split a couple of things from it into their own separate PRs first.

**Comment by penev92** [comment]
And here we are, almost 5 months later, back to this with the last 3 commits of the series. :)

File: OpenRA.Game/GameInformation.cs [hunk 2]
```diff
@@ -147,10 +147,10 @@ public class Player
 			public bool IsHuman;
 			public bool IsBot;
 
-			// The faction name (aka Country)
+			// The faction name.
```
Comment (Phrohdoh) [inline OpenRA.Game/GameInformation.cs]: Please note that this is the display name.

**Comment by abcdefg30** [comment]
:+1:

**Pull request merged**